### PR TITLE
Pb 2353 contributor picture cropping

### PR DESCRIPTION
**Pull request by richard015ar**
Related issue: https://github.com/pressbooks/pressbooks/issues/2353

This change fixes the WP media cropping settings for contributors' picture feature.
When the image is taller than larger and it's more than 400px of larger, the cropper fails due a pixel falls out of the image.
Previously there is not a check for minimal size (400px), there was just a suggestion. Now the the checking is performed from the back end and front end sides. Now, images smaller then 400px are not allowed to upload. If a image smaller than 400px is selected from the media library, it is not inserted for the picture and a error is displayed to inform the user.

### Test
- Add or edit a contributor in a book (Book Info > Contributors)
- Add an image larger than 400px and taller than larger. The cropper should work.
- Try to upload a image smaller than 400px (width and/or height). A message should be displayed indicating the minimum size is 400px.
- Try to add an image smaller than 400px from the media library (instead uploading from the contributors section)
- Try to add that image in any contributor's profile picture. An error message must be displayed.

### CODE REVIEW

File: assets/src/scripts/contributors.js [hunk 1]
```diff
@@ -164,6 +164,9 @@ jQuery( function ( $ ) {
 		} );
 		pictureLibrary.on( 'select', function () {
 			const attachment = pictureLibrary.state().get( 'selection' ).first().toJSON();
+			if ( attachment.width < minPictureSize || attachment.height < minPictureSize ) {
+				return;
```
Comment (richard015ar) [inline assets/src/scripts/contributors.js:168]: here an alert would be possible before closing the wp media

File: inc/admin/metaboxes/namespace.php [hunk 2]
```diff
@@ -1223,7 +1223,7 @@ function contributor_add_form() {
 					<button name="dispatch-media-picture" id="btn-media">Upload Picture</button>
 					<input type="hidden" name="<?php echo $term; ?>" id="<?php echo $meta_tags['tag']; ?>">
 					<p>
-						<?php echo __( 'Images should be square and at least 400px wide. Very large images will be resized upon upload.', 'pressbooks' ); ?>
+						<?php echo __( 'Images should be square (400px x 400px). You will be allowed to crop images after upload. The maximum file size permitted is 1 MB.', 'pressbooks' ); ?>
```
Comment (richard015ar) [inline inc/admin/metaboxes/namespace.php:1226]: Confirm this with Steel

File: assets/src/scripts/contributors.js [hunk 3]
```diff
@@ -77,9 +77,9 @@ jQuery( function ( $ ) {
 			minWidth: xImg > xInit ? xInit : xImg,
 			minHeight: yImg > yInit ? yInit : yImg,
 			x1: x1,
```
Comment (richard015ar) [inline assets/src/scripts/contributors.js:79]: Here we can use `maxHeight` and `maxWidth` if we want to limit the cropped image resolution to a specific number of pixel. Ask to Steel.

**Review by fdalcin — APPROVED** [review]
@richard015ar LGTM. 

- [x] Images smaller than 400px are not allowed to be uploaded nor selected. 
- [x] Images that are greater than 400px show a cropping option with a minimum size of 400px.
- [x] Images that are taller than wider are now showing the cropping option.

**Pull request merged**